flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/benchmark/StreamNetworkBenchmarkEnvironment.java

```diff
@@ -227,39 +229,22 @@ protected ResultPartitionWriter createResultPartition(
 	}
 
 	private InputGate createInputGate(
-			IntermediateDataSetID dataSetID,
-			ExecutionAttemptID executionAttemptID,
 			final TaskManagerLocation senderLocation,
-			NetworkEnvironment environment,
+			final SingleInputGateFactory gateFactory,
```

Comment:
is `gateFactory` not already available as a class field in the method?
strange that the field is protected, can be private

Reply:
I also found this issue before, but considering all the fields in this class are defined as protected, so I keep the way as others. 
Also for the class field factory, I keep the way as field `channels`  to pass because it is not initialized in the constructor. Actually I also think many other fields in this class could not be passed as well.
Do you think we should further refactoring this?